tvm/serialization/boc.mdx

```diff
@@ -2,6 +2,279 @@
+and up to six binary `0s` are appended to the data bits. After that, the data is split into $\lceil\frac{b}{8}\rceil$ 8-bit groups.
+
+```json
+01     -> 01100000 = 0x60
+fe     -> do not change (full groups)
+0aaaaa -> do not change (full groups)
+```
+
+Next come the depths for the refs in two bytes:
+
+```json
+01     -> 0002
+fe     -> 0001
+0aaaaa -> 0000
+```
+
+Now specify which cells each one references:
+
+```json
+0: 01     -> 0201: refers to 2 cells with such indexes
+1: fe     -> 02: refers to cells with index 2
+2: 0aaaaa -> no refs
+```
+
+For each cell we have its hexadecimal representation:
+
+```json
+01     -> 02016000020201
+fe     -> 0102fe000102
+0aaaaa -> 00060aaaaa0000
+```
+
+Finally, we concatenate all parts into a single hexadecimal array:
+`0x020160000202010102fe00010200060aaaaa0000`.
+
+Now that we've serialized our cells into a flat 20-byte array, it's time to pack them into a complete BoC format.
+
+```text
+0xb5ee9c72                                 -> TL-B id of the BoC structure
+0b1                                        -> has indexes
+0b0                                        -> does not have CRC32C
+0b0                                        -> does not have cache bits
+0b00                                       -> flags are 0
+0b001                                      -> the number of bytes needed to store the number of cells is 1
+0b00000001                                 -> the number of bytes used to represent offset of a serialization is 1
+0b00000011                                 -> the number of cells is 3
+0b00000001                                 -> the number of roots is 1
+0b00000000                                 -> the number of absent cells is 0
+0b00010100                                 -> tot_cells_size is 20 bytes
+0b00000000                                 -> the root list; we have one root with number 0 after the topological sort
+0b000001110000111000010100                 -> the three 8-bits group of indexes for cells accorging to the topological sort
+0x020160000202010102fe00010200060aaaaa0000 -> the cells data
+_                                          -> CRC32C is not serialized
+```
+
+By combining everything into a single bit string, we get the result of serialization.
+
+## SDK from @ton/core
+
+According to the TL-B scheme above there is the [SDK](https://github.com/ton-org/ton-core/blob/4577e94/src/boc/cell/serialization.ts#L1)
+for serialization and parsing BoC.
+
+Only serialization of BoCs with one root and no absent cells is supported.
+
+```typescript
+import { beginCell, serializeBoc, deserializeBoc } from "@ton/core";
+// serializeBoc has two arguments:
+// root: Cell. A root cell of a given tree of cells
+// opt: { idx: boolean, crc32: boolean }. Two flags indicating whether indexes and CRC32C will be included in serialization
+// the output is a Buffer with serialization
```

Comment:
this explanation should be moved out of the code block and explained in the regular text